libfwupdplugin/fu-archive.c

```diff
@@ -256,3 +407,29 @@ fu_archive_new(GBytes *data, FuArchiveFlags flags, GError **error)
 		return NULL;
 	return g_steal_pointer(&self);
 }
+
+/**
+ * fu_archive_get_bytes:
+ * @callback: a #FuArchiveEntryIterateFunc.
+ * @user_data: user data
+ * @error: (nullable): optional return location for an error
+ *
+ * TODO
+ *
+ * Since: 1.8.1
+ **/
+GBytes *
+fu_archive_get_bytes(FuArchive *self,
```

Comment:
not sure about the name. perhaps fu_archive_write() or fu_archive_save_as_bytes()

Reply:
why not